File: 55_Jump_Game.py
class Solution:
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        _size = len(nums)
        
        #print(_size)
        
        lastReachable = _size - 1;
        for i in range(_size-1, -1, -1):
            if(i + nums[i] >= lastReachable):
                lastReachable = i
            
           # print(lastReachable)
        
        return lastReachable == 0

File: 55_Jump_game.cpp
//55. Jump Game
//https://leetcode.com/problems/jump-game/description/



class Solution {
public:
    bool canJump(vector<int>& nums) {
        
        int next = 0, _size = nums.size();
        int lastReachable = _size-1;
        
        for(int i= _size-1; i>=0; i--){
            if(i + nums[i] >= lastReachable)
                lastReachable = i;
        }
        
        return lastReachable == 0;
    }
};
